nengo/neurons.py

```diff
@@ -34,19 +34,37 @@ def _argreprs(self):
     def current(self, x, gain, bias):
         """Compute current injected in each neuron given input, gain and bias.
 
+        Note that ``x`` is assumed to be already projected onto the encoders
+        associated with the neurons and normalized to radius 1, so the maximum
+        expected current for a neuron occurs when input for that neuron is 1.
+
         Parameters
         ----------
-        x : (n_neurons,) array_like
-            Vector-space input.
+        x : (n_samples,) or (n_samples, n_neurons) array_like
+            Scalar inputs for which to calculate current.
         gain : (n_neurons,) array_like
             Gains associated with each neuron.
         bias : (n_neurons,) array_like
             Bias current associated with each neuron.
+
+        Returns
+        -------
+        current : (n_samples, n_neurons)
+            Current to be injected in each neuron.
+            If ``n_samples`` is 1, the extra dimension will be removed.
         """
         x = np.array(x, dtype=float, copy=False, ndmin=1)
         gain = np.array(gain, dtype=float, copy=False, ndmin=1)
         bias = np.array(bias, dtype=float, copy=False, ndmin=1)
-        return gain * x + bias
+
+        if x.ndim == 1:
+            x = x[:, np.newaxis]
+        elif x.ndim > 1 and x.shape[1] != gain.shape[0]:
+            raise ValidationError("Expected shape (%d, %d); got (%d, %d)."
+                                  % ((x.shape[0], gain.shape[0]) + x.shape),
+                                  attr="x", obj=self)
+
+        return np.squeeze(gain * x + bias)
```

Comment:
I don't like the `squeeze`. Often, I work explicitly with a certain number (e.g. 2) dimensions, and if `n_neurons` or `n_samples` happens to be 1, I don't want to lose those dimensions. This could potentially cause bugs if e.g. we have an ensemble with 1 neuron.

Reply:
I agree that I'd prefer to avoid the squeeze. If you do a ctrl+F in the main PR thread you'll see a bunch of discussion on this and the various cases.

Reply:
This is done in option 3.